sljit_src/sljitNativeLOONGARCH_64.c

```diff
@@ -805,16 +805,14 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_has_cpu_feature(sljit_s32 feature_type)
 	case SLJIT_HAS_SIMD:
 		return (LOONGARCH_HWCAP_LSX & get_cpu_features(GET_HWCAP));
 
-	case SLJIT_HAS_ATOMIC:
-		return (LOONGARCH_CFG2_LAMCAS & get_cpu_features(GET_CFG2));
-
 	case SLJIT_HAS_CLZ:
 	case SLJIT_HAS_CTZ:
 	case SLJIT_HAS_REV:
 	case SLJIT_HAS_ROT:
 	case SLJIT_HAS_PREFETCH:
 	case SLJIT_HAS_COPY_F32:
 	case SLJIT_HAS_COPY_F64:
+	case SLJIT_HAS_ATOMIC:
```

Comment:
Am I see it correctly, that the cpu has new LL/SC form but CAS form is also supported? This is true for ARM64, so I am thinking whether we should allow users to pick which form they prefer. E.g. add a flag to pick the type, then return with unsupported if not available.

Reply:
LoongArch originally only support LL/SC, but add CAS support on LoongArch V1.10 (Currently only 3A6000 support this extension, QEMU do not support it yet). So it's safe to always use LL/SC, but CAS give more fine-grained atomic operations support. 

To allow user's selection, maybe we should add a type param to `sljit_emit_atomic_load` or `sljit_emit_atomic_store`, or we could add a `SLJIT_HAS_CAS` flag for it? But I still think there is not too much need to offer choices. Unless CAS is slower that LL/SC on some specific platform.

Reference: https://loongson.github.io/LoongArch-Documentation/LoongArch-Vol1-EN.html#new-in-loongarch-v1.1

Reply:
Update: AMCAS is faster than LL/SC on LoongArch

Reply:
Ok. I will think about it.